pkg/webui/console/views/device-general-settings/network-server-form/index.js

```diff
@@ -0,0 +1,293 @@
+      const isABP = initialValues._activation_mode === ACTIVATION_MODES.ABP
+      const isOTAA = initialValues._activation_mode === ACTIVATION_MODES.OTAA
+
+      const castedValues = validationSchema.cast(values)
+      const updatedValues = diff(initialValues, castedValues, [
+        '_activation_mode',
+        '_external_js',
+        '_joined',
+      ])
+
+      if (isABP) {
+        // Do not reset session keys
+        if (updatedValues.session.keys && Object.keys(updatedValues.session.keys).length === 0) {
+          delete updatedValues.session.keys
+        }
+
+        if (Object.keys(updatedValues.session).length === 0) {
+          delete updatedValues.session
+        }
+      } else if (isOTAA) {
+        if (updatedValues.root_keys && Object.keys(updatedValues.root_keys).length === 0) {
+          delete updatedValues.root_keys
+        }
+      }
+
+      setError('')
+      try {
+        await onSubmit(updatedValues)
+        resetForm(castedValues)
+        onSubmitSuccess()
+      } catch (err) {
+        setSubmitting(false)
+        setError(err)
+      }
+    },
+    [initialValues, onSubmit, onSubmitSuccess],
+  )
+
+  const handleResetsFCntChange = React.useCallback(evt => {
+    const { checked } = evt.target
+
+    setResetsFCnt(checked)
+  }, [])
+
+  const handleVersionChange = React.useCallback(
+    version => {
+      const isABP = initialValues._activation_mode === ACTIVATION_MODES.ABP
+      const lwVersion = parseLorawanMacVersion(version)
+      setLorawanVersion(lwVersion)
+      const { setValues, state: formState } = formRef.current
+      const { session = {} } = formState.values
+      const { session: initialSession } = initialValues
+      if (lwVersion >= 110) {
+        const updatedSession = isABP
+          ? {
+              dev_addr: session.dev_addr,
+              keys: {
+                ...session.keys,
+                s_nwk_s_int_key:
+                  session.keys.s_nwk_s_int_key || initialSession.keys.s_nwk_s_int_key,
+                nwk_s_enc_key: session.keys.nwk_s_enc_key || initialSession.keys.nwk_s_enc_key,
+              },
+            }
+          : session
+        setValues({
+          ...formState.values,
+          lorawan_version: version,
+          session: updatedSession,
+        })
+      } else {
+        const updatedSession = isABP
+          ? {
+              dev_addr: session.dev_addr,
+              keys: {
+                f_nwk_s_int_key: session.keys.f_nwk_s_int_key,
+              },
+            }
+          : session
+        setValues({
+          ...formState.values,
+          lorawan_version: version,
+          session: updatedSession,
+        })
+      }
+    },
+    [initialValues],
+  )
+
+  return (
+    <Form
+      validationSchema={validationSchema}
+      initialValues={initialValues}
+      onSubmit={onFormSubmit}
+      error={error}
+      formikRef={formRef}
+      enableReinitialize
```

Comment:
Please add a field description for MAC/PHY/FP as well.
@htdvisser can you help out with wording?